src/main/java/edu/ucsb/cs156/happiercows/controllers/UserCommonsController.java

```diff
@@ -57,4 +62,42 @@ public UserCommons getUserCommonsById(
     return userCommons;
   }
 
+  @ApiOperation(value = "Update user commons when cow is bought")
+  @PreAuthorize("hasRole('ROLE_USER')")
+  @PostMapping("/buy")
+  public UserCommons updateWhenBuy(
+      @ApiParam("commonsId") @RequestParam Long commonsId) throws JsonProcessingException {
+
+    User u = getCurrentUser().getUser();
+    Long userId = u.getId();
+    UserCommons userCommons = userCommonsRepository.findByCommonsIdAndUserId(commonsId, userId)
+        .orElseThrow(
+            () -> new EntityNotFoundException(UserCommons.class, "commonsId", commonsId, "userId", userId));
+    userCommons.setCowHealth((userCommons.getNumCows() * userCommons.getCowHealth() + 1) / (userCommons.getNumCows() + 1)); // need to set precision on frontend!
+    userCommons.setNumCows(userCommons.getNumCows() + 1);
+    userCommons.setTotalWealth(userCommons.getTotalWealth() - userCommons.getCowPrice());
+    userCommonsRepository.save(userCommons);
+    return userCommons;
+  }
+
+
+  @ApiOperation(value = "Update user commons when cow is sold")
+  @PreAuthorize("hasRole('ROLE_USER')")
+  @PutMapping("/sell")
+  public UserCommons updateWhenSell(
+      @ApiParam("commonsId") @RequestParam Long commonsId) throws JsonProcessingException {
+
+    User u = getCurrentUser().getUser();
+    Long userId = u.getId();
+    UserCommons userCommons = userCommonsRepository.findByCommonsIdAndUserId(commonsId, userId)
+        .orElseThrow(
+            () -> new EntityNotFoundException(UserCommons.class, "commonsId", commonsId, "userId", userId));
+    
+    userCommons.setNumCows(userCommons.getNumCows() - 1);
+    userCommons.setTotalWealth(userCommons.getTotalWealth() + (userCommons.getCowPrice() * .8 * (userCommons.getCowHealth()/100)));
```

Comment:
It feels like this formula, i.e. the formula for selling price of a cow, should be somewhere in a class so that it appears only once in the code base.

I'll put that in a future issue though.

Reply:
Nah, changed my mind.  I see that you added `getCowPrice()` and `getCowHealth()` to `UserCommons`.   I suggest that you also add this:

```
public double getCowSellingPrice() {
  return (this.getCowPrice() * .8 * (this.getCowHealth()/100)));
}
```

Along with adding a unit test in a new file UserCommonsTests.java (under a new directory `/src/test/java/edu/ucsb/cs156/happiercows/entities`) for just getCowSellingPrice.

Then change the line 
```
userCommons.setTotalWealth(userCommons.getTotalWealth() + (userCommons.getCowPrice() * .8 * (userCommons.getCowHealth()/100)));
```

to 
```
userCommons.setTotalWealth(userCommons.getTotalWealth() + userCommons.getCowSellingPrice());
```